test/parallel-benchmark/mzcompose.py

```diff
@@ -105,24 +107,50 @@ def known_regression(scenario: str, other_tag: str) -> bool:
 
 
 class Statistics:
-    def __init__(self, times: list[float], durations: list[float]):
-        assert len(times) == len(durations)
-        self.queries: int = len(times)
-        self.qps: float = len(times) / max(times)
-        self.max: float = max(durations)
-        self.min: float = min(durations)
-        self.avg: float = float(numpy.mean(durations))
-        self.p50: float = float(numpy.median(durations))
-        self.p95: float = float(numpy.percentile(durations, 95))
-        self.p99: float = float(numpy.percentile(durations, 99))
-        self.p99_9: float = float(numpy.percentile(durations, 99.9))
-        self.p99_99: float = float(numpy.percentile(durations, 99.99))
-        self.p99_999: float = float(numpy.percentile(durations, 99.999))
-        self.p99_9999: float = float(numpy.percentile(durations, 99.9999))
-        self.p99_99999: float = float(numpy.percentile(durations, 99.99999))
-        self.p99_999999: float = float(numpy.percentile(durations, 99.999999))
-        self.std: float = float(numpy.std(durations, ddof=1))
-        self.slope: float = float(numpy.polyfit(times, durations, 1)[0])
+    def __init__(self, action: str, m: MeasurementsStore, start_time: float):
+        if isinstance(m, MemoryStore):
+            times: list[float] = [x.timestamp - start_time for x in m.data[action]]
+            durations: list[float] = [x.duration * 1000 for x in m.data[action]]
+            self.queries: int = len(times)
+            self.qps: float = len(times) / max(times)
+            self.max: float = max(durations)
+            self.min: float = min(durations)
+            self.avg: float = float(numpy.mean(durations))
+            self.p50: float = float(numpy.median(durations))
+            self.p95: float = float(numpy.percentile(durations, 95))
+            self.p99: float = float(numpy.percentile(durations, 99))
+            self.p99_9: float = float(numpy.percentile(durations, 99.9))
+            self.p99_99: float = float(numpy.percentile(durations, 99.99))
+            self.p99_999: float = float(numpy.percentile(durations, 99.999))
+            self.p99_9999: float = float(numpy.percentile(durations, 99.9999))
+            self.p99_99999: float = float(numpy.percentile(durations, 99.99999))
+            self.p99_999999: float = float(numpy.percentile(durations, 99.999999))
+            self.std: float = float(numpy.std(durations, ddof=1))
+            self.slope: float = float(numpy.polyfit(times, durations, 1)[0])
+        elif isinstance(m, SQLiteStore):
+            cursor = m.conn.cursor()
+            cursor.execute(
+                f"SELECT count(*), count(*) / (max(timestamp) - {start_time}), max(duration) * 1000, min(duration) * 1000, avg(duration) * 1000 FROM measurements WHERE scenario = ? AND action = ?",
+                (m.scenario, action),
+            )
+            self.queries, self.qps, self.max, self.min, self.avg = cursor.fetchone()
+            # TODO: Rest
```

Comment:
There is an open TODO. Do you plan to address this within this PR?

Note that I cancelled auto-merge. Feel free to proceed, though.

Reply:
Nope, I left that for future me.